editoast/src/views/infra/auto_fixes/operational_point.rs

```diff
@@ -1,26 +1,86 @@
+use editoast_schemas::primitives::OSRDIdentified;
 use editoast_schemas::primitives::OSRDObject as _;
 use editoast_schemas::primitives::ObjectRef;
+use editoast_schemas::primitives::ObjectType;
+use itertools::Itertools;
+use json_patch::Patch;
+use json_patch::PatchOperation;
+use json_patch::RemoveOperation;
 use std::collections::HashMap;
-
 use tracing::debug;
 
-use super::new_ref_fix_delete_pair;
 use super::Fix;
+use super::OrderedOperation;
 use crate::generated_data::infra_error::InfraError;
 use crate::generated_data::infra_error::InfraErrorType;
 use crate::infra_cache::object_cache::OperationalPointCache;
+use crate::infra_cache::operation::CacheOperation;
+use crate::infra_cache::operation::DeleteOperation;
+use crate::infra_cache::operation::Operation;
+use crate::infra_cache::operation::UpdateOperation;
+use crate::infra_cache::ObjectCache;
+
+fn invalid_reference_to_ordered_operation(
+    operational_point: &OperationalPointCache,
+    object_ref: &ObjectRef,
+) -> Option<OrderedOperation> {
+    let (track_range_idx, _) = operational_point
+        .parts
+        .iter()
+        .enumerate()
+        .find(|(_idx, part)| part.track.as_str() == object_ref.obj_id)?;
+    Some(OrderedOperation::RemoveTrackRef { track_range_idx })
+}
 
 pub fn fix_operational_point(
     operational_point: &OperationalPointCache,
     errors: impl Iterator<Item = InfraError>,
 ) -> HashMap<ObjectRef, Fix> {
-    errors
+    let mut new_op = operational_point.clone();
+    let operation = errors
         .filter_map(|infra_error| match infra_error.get_sub_type() {
-            InfraErrorType::EmptyObject => Some(new_ref_fix_delete_pair(operational_point)),
+            InfraErrorType::EmptyObject => Some(OrderedOperation::Delete),
+            InfraErrorType::InvalidReference { reference }
+                if reference.obj_type == ObjectType::TrackSection =>
+            {
+                new_op
+                    .parts
+                    .retain(|part| part.track.as_str() != reference.obj_id);
```

Comment:
I have a problem understanding why you need to create this `new_op` and modify it? Especially, I don't understand the difference with `electrification.rs` or `speed_section.rs` where we do not do this. Should fix those too?